contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortSwapJob.java

```diff
@@ -98,6 +104,16 @@ public void run() throws Exception {
         JobConf conf = new JobConf();
         Path dataPath = new Path(dataDir);
         String modifiedDataDir = dataPath.makeQualified(FileSystem.get(conf)).toString();
+        boolean cdnEnabled = props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_ENABLED, false);
+
+        if (cdnEnabled) {
+            if (modifiedDataDir.matches(".*hdfs://.*:[0-9]{1,5}/.*")) {
+                new GobblinDistcpJob(getId(), modifiedDataDir, clusterName, props).run();
+                modifiedDataDir = GobblinDistcpJob.source;
```

Comment:
This logic seems wired.
Can we define a function to return the final data dir in GobblinDistcpJob?